file_writer_control/WorkerFinder.py

```diff
@@ -0,0 +1,118 @@
+from file_writer_control.WorkerStatus import WorkerStatus
+from file_writer_control.WriteJob import WriteJob
+from file_writer_control.JobStatus import JobState, JobStatus
+from file_writer_control.CommandStatus import CommandStatus
+from typing import List
+from file_writer_control.CommandChannel import CommandChannel
+from file_writer_control.CommandHandler import CommandHandler
+from kafka import KafkaProducer
+from file_writer_control.KafkaTopicUrl import KafkaTopicUrl
+from datetime import datetime
+from streaming_data_types.run_stop_6s4t import serialise_6s4t as serialise_stop
+from file_writer_control.CommandId import generate_command_id
+
+
+class WorkerFinderBase:
+    def __init__(self, command_topic: str, command_channel: CommandChannel, message_producer: KafkaProducer):
+        """
+        Constructor.
+        """
+        self.command_channel = command_channel
+        self.command_topic = command_topic
+        self.message_producer = message_producer
+
+    def send_command(self, message: bytes):
+        """
+        Send a message (command) to the file-writer "command"-topic.
+        :param message: The command/message as binary data.
+        """
+        self.message_producer.send(self.command_topic, message)
+
+    def try_start_job(self, job: WriteJob) -> CommandHandler:
```

Comment:
So If I am going to implement it in the child, what is the need having it here? I mean there is nothing to override? May be I am missing something, could you elaborate?

Reply:
There are a few reasons. The most important one is that as the presence of this member function is required by other classes, it really should be documented somewhere. Better to document it here than to have it in some README-document.